contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {RouterBase, ITypeAndVersion} from "./RouterBase.sol";
+import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
+import {IFunctionsCoordinator} from "./interfaces/IFunctionsCoordinator.sol";
+import {IFunctionsSubscriptions, FunctionsSubscriptions, IFunctionsBilling} from "./FunctionsSubscriptions.sol";
+
+contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions {
+  event RequestStart(bytes32 indexed requestId, Request request);
+  event RequestEnd(
+    bytes32 indexed requestId,
+    uint64 indexed subscriptionId,
+    uint96 totalCostJuels,
+    address transmitter,
+    uint8 resultCode,
+    bytes response
+  );
+
+  error OnlyCallableFromCoordinator();
+
+  struct CallbackResult {
+    bool success;
+    uint256 gasUsed;
+  }
+
+  // ================================================================
+  // |                    Configuration state                       |
+  // ================================================================
+  struct Config {
+    // Flat fee (in Juels of LINK) that will be paid to the Router owner for operation of the network
+    uint96 adminFee;
+    // The function selector that is used when calling back to the Client contract
+    bytes4 handleOracleFulfillmentSelector;
+  }
+  Config private s_config;
+  event ConfigSet(uint96 adminFee, bytes4 handleOracleFulfillmentSelector);
+
+  error OnlyCallableByRoute();
+
+  // ================================================================
+  // |                       Initialization                         |
+  // ================================================================
+  constructor(
+    uint16 timelockBlocks,
+    uint16 maximumTimelockBlocks,
+    address linkToken,
+    bytes memory config
+  ) RouterBase(msg.sender, timelockBlocks, maximumTimelockBlocks, config) FunctionsSubscriptions(linkToken) {}
+
+  /**
+   * @inheritdoc ITypeAndVersion
+   */
+  function typeAndVersion() public pure override returns (string memory) {
+    return "Functions Router v1";
+  }
+
+  // ================================================================
+  // |                 Configuration methods                        |
+  // ================================================================
+  /**
+   * @notice Sets the configuration for FunctionsRouter specific state
+   * @param config bytes of config data to set the following:
+   *  - adminFee: fee that will be paid to the Router owner for operating the network
+   */
+  function _setConfig(bytes memory config) internal override {
```

Comment:
Do we plan to have a callable (non-internal) version of this function? 
https://www.notion.so/chainlink/Set-the-DON-s-OCR-configuration-931138ad504843e29a49dda6473f3bd2?pvs=4

Reply:
This is part of our upgradability pattern.  See `RouterBase` for details, but `updateConfig()` from `RouterBase` is the external function used to call `_setConfig`

Reply:
The callable path is to go through the Router's `proposeConfig` for itself (empty bytes for the id)

Reply:
synced offline